release-24.1: sql/distsql: preserve JobID in redacted logs

spilchen opened the pull request:
Backport 1/1 commits from #146362.

/cc @cockroachdb/release

---

Previously, the JobID log tag could be redacted, making it difficult to trace TTL job execution when viewing redacted logs. This change ensures the JobID is logged as a redact.SafeString, allowing it to appear in redacted outputs for easier debugging.

Additionally, the flowCtx.ID is now also saved as a safe string too.

Epic: none
Release note: none

Release justification: improves job observability

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146469)
<!-- Reviewable:end -->

spilchen commented:
fyi, the original PR unredeacted the flow ID. But I left it off of this backport because there were a couple of changes that it depended on that weren’t backported. On this branch, the code to include the flow ID in the log tag is as follows:

```
flowCtx.AmbientContext.AddLogTag("f", flowCtx.ID.Short())
```
  
Where `Short()` is defined as:
 
```
// Short returns the first eight characters of the output of String().
func (u UUID) Short() string {
	return u.String()[:8]
}
```
    
Where as, in 25.1 and newer, `Short()` is defined as:

```
// Short returns an abbreviated version of the UUID containing the first four
// bytes.
func (u UUID) Short() Short {
	return Short{
		b: [shortSize]byte(u[0:shortSize]),
	}
}
``` 

So, I would need to backport #136636 to get the above `Short()` method and #141390 to unredact the Short type.

spilchen committed:
sql: preserve JobID in redacted logs

Previously, the JobID log tag could be redacted, making it difficult to
trace TTL job execution when viewing redacted logs. This change ensures
the JobID is logged as a redact.SafeString, allowing it to appear in
redacted outputs for easier debugging.

Additionally, the flowCtx.ID is now also saved as a safe string too.

Epic: none
Release note: none
